adapters/src/main/java/io/realm/RealmBaseRecyclerAdapter.java

```diff
@@ -0,0 +1,164 @@
+        if (context == null) {
+            throw new IllegalArgumentException("Context can not be null");
+        }
+
+        this.context = context;
+        this.adapterData = data;
+        this.inflater = LayoutInflater.from(context);
+        this.hasAutoUpdates = autoUpdate;
+        this.listener = hasAutoUpdates ? new RealmChangeListener<BaseRealm>() {
+            @Override
+            public void onChange(BaseRealm results) {
+                notifyDataSetChanged();
+            }
+        } : null;
+    }
+
+    @Override
+    public void onAttachedToRecyclerView(final RecyclerView recyclerView) {
+        super.onAttachedToRecyclerView(recyclerView);
+        if (hasAutoUpdates && isDataValid()) {
+            addListener(adapterData);
+        }
+    }
+
+    @Override
+    public void onDetachedFromRecyclerView(final RecyclerView recyclerView) {
+        super.onDetachedFromRecyclerView(recyclerView);
+        if (hasAutoUpdates && isDataValid()) {
+            removeListener(adapterData);
+        }
+    }
+
+    /**
+     * Returns the current ID for an item. Note that item IDs are not stable so you cannot rely on the item ID being the
+     * same after notifyDataSetChanged() or {@link #updateData(OrderedRealmCollection)} has been called.
+     *
+     * @param index position of item in the adapter.
+     * @return current item ID.
+     */
+    @Override
+    public long getItemId(final int index) {
+        return index;
+    }
+
+    @Override
+    public int getItemCount() {
+        return isDataValid() ? adapterData.size() : 0;
+    }
+
+    /**
+     * Returns the item associated with the specified position.
+     * Can return null if provided Realm instance by {@link OrderedRealmCollection} is closed.
+     *
+     * @param index index of the item.
+     * @return the item at the specified position, null if adapter data is not valid.
+     */
+    public T getItem(int index) {
+        return isDataValid() ? adapterData.get(index) : null;
+    }
+
+    /**
+     * Returns data associated with this adapter.
+     *
+     * @return adapter data.
+     */
+    public OrderedRealmCollection<T> getData() {
+        return adapterData;
+    }
+
+    /**
+     * Updates the data associated to the Adapter. Useful when the query has been changed.
+     * If the query does not change you might consider using the automaticUpdate feature.
+     *
+     * @param data the new {@link OrderedRealmCollection} to display.
+     */
+    public void updateData(OrderedRealmCollection<T> data) {
+        if (hasAutoUpdates) {
+            if (adapterData != null) {
+                removeListener(adapterData);
+            }
+            if (data != null) {
+                addListener(data);
+            }
+        }
+
+        this.adapterData = data;
+        notifyDataSetChanged();
+    }
+
+    private void addListener(OrderedRealmCollection<T> data) {
+        BaseRealm realm = getRealm(data);
```

Comment:
Since RecyclerView has proper lifecycle methods you can actually use the logic I accidentially added to RealmBaseAdapter here: https://github.com/realm/realm-android-adapters/blob/e395cb11b36f483dfa35f0b9d150f3d3c94e67ff/adapters/src/main/java/io/realm/RealmBaseAdapter.java

Because with a proper remove method you can safely use better listener on `RealmResults` instead of registering on Realm. That way we avoid updating the RecyclerView on any changes to Realm when using RealmResults.